### DISCUSSION TITLE
Question: should I be able to redirect link/script tags in the <head>?

### DISCUSSION THREAD

**Discussion by BMCwebdev** [mswjs/msw#875] [Q&A]
Just getting started, and have been looking around for an answer on this one. Should MSW be able to redirect a GET for a `<link>` or `<script>` tag in the `<head>`? Something along the lines of:
`  rest.get("/styles.css", (req, res, ctx) => {
    return res(
      ctx.status(307),
      ctx.set("Location", "/test/styles.css")
    );
  })`
I forked a demo CodeSandbox, added a link tag in the index.html' s `<head>` and tried the above code without luck. Link is below:
https://codesandbox.io/s/hardcore-benz-4bw1b

**Comment by kettanaito** [comment]
Hey, @BMCwebdev. 

Although Service Workers can capture such static assets on page load, MSW is a little different. Our worker self-destructs when its last client unloads. This includes the case when you refresh the page. What actually happens is that the worker unregisters itself when the page unloads, then your page loads your JavaScript code that calls `worker.start()`, which signals the worker to activate again. 

There's a loading gap between the page load and its JavaScript being executed. That's one of the reasons why MSW cannot intercept the requests from the `<head>`. There's also another reason. 

We are explicitly bypassing any initial load assets: 

https://github.com/mswjs/msw/blob/9373ccf5e699fd46406aadbb55580b4a833e4541/src/mockServiceWorker.js#L261-L266

https://github.com/mswjs/msw/blob/9373ccf5e699fd46406aadbb55580b4a833e4541/src/mockServiceWorker.js#L151-L157

We do so that you couldn't mock your `index.html`, or other cricial assets, by mistake, resulting in a broken application until you manually unregister the worker. I find this a reasonable default and have no intention of changing this behavior.

MSW is primarily meant for _interactive_ requests. Basically, any `fetch` or other requests you do on runtime. It can still intercept requests on page load, but those must also be issued via `fetch`.
